Xamarin.Forms.Core/Application.cs

```diff
@@ -26,6 +26,25 @@ public class Application : Element, IResourcesProvider, IApplicationController,
 		[Obsolete("Assign the LogWarningsListener")]
 		public static bool LogWarningsToApplicationOutput { get; set; }
 
+		private static Action<(string Category, string Message)> CurrentLogWarningListener;
+		private static DelegateLogListener CurrentLogListener;
+
+		public static Action<(string Category, string Message)> LogWarningsListener
+		{
+			get => CurrentLogWarningListener;
+
+			set
+			{
+				if (CurrentLogListener != null && Log.Listeners.Contains(CurrentLogListener))
+					Log.Listeners.Remove(CurrentLogListener);
+
+				CurrentLogWarningListener = value;
+				CurrentLogListener = new DelegateLogListener((category, message) => CurrentLogWarningListener.Invoke((category, message)));
```

Comment:
What happens if I set `LogWarningsListener` to `null`?

Reply:
Ha! Nice. Just tested it, nothing really did happen. Added a check for it anyway.